flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/utils/SourceCodeGenerator.java

```diff
@@ -43,6 +44,84 @@ public String generateCodeSnippetForType(TypeData typeData) {
+        if (nodeKind != NodeKind.CLASS) {
+            return "";
+        }
+
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("service class ")
+                .append(typeData.name())
+                .append(" {");
+
+        // Add inferred fields from functions
+        for (Function function : typeData.functions()) {
+            generateInferredGraphqlClassField(function, stringBuilder);
+        }
+
+        // init functions
+        stringBuilder.append("\n\t")
+                .append("function init(");
+        for (int i = 0; i < typeData.functions().size(); i++) {
+            Function function = typeData.functions().get(i);
+            generateTypeDescriptor(function.returnType(), stringBuilder);
+            stringBuilder.append(" ").append(function.name());
+            if (i < typeData.functions().size() - 1) {
+                stringBuilder.append(", ");
+            }
+        }
+        stringBuilder.append(") {");
+        for (int i = 0; i < typeData.functions().size(); i++) {
+            Function function = typeData.functions().get(i);
+            stringBuilder.append("\n\t\tself.")
+                    .append(function.name())
+                    .append(" = ")
+                    .append(function.name()).append(";");
+        }
+        stringBuilder.append("\n\t}");
+
+        // Add resource functions
+        for (Function function : typeData.functions()) {
+            if (function.description() != null && !function.description().isEmpty()) {
+                stringBuilder
+                        .append("\n\t")
+                        .append(CommonUtils.convertToBalDocs(function.description()));
+            } else {
+                stringBuilder.append("\n");
+            }
+            stringBuilder.append("\t")
+                    .append("resource function ")
+                    .append(function.accessor())
+                    .append(" ")
+                    .append(function.name())
+                    .append("(");
+
+            // Function params
+            for (Member param: function.parameters()) {
+                generateTypeDescriptor(param.type(), stringBuilder);
+                stringBuilder.append(" ").append(param.name());
+                if (param.defaultValue() != null && !param.defaultValue().isEmpty()) {
+                    stringBuilder.append(" = ").append(param.defaultValue());
+                }
+                if (function.parameters().indexOf(param) < function.parameters().size() - 1) {
+                    stringBuilder.append(", ");
+                }
+            }
+
+            stringBuilder.append(") returns ");
+            generateTypeDescriptor(function.returnType(), stringBuilder);
+            stringBuilder.append(" {");
+
+            // Function body: "return self.<function-name>;"
+            stringBuilder.append("\n\t\treturn self.").append(function.name()).append(";");
```

Comment:
in here the function body should contain do clause with error handler

Reply:
That make sense. WDYT @hasithaa

Reply:
Had a offline discussion on this. We thought of having this. Will update in a new PR.